flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/sql/JoinTest.scala

```diff
@@ -55,12 +55,13 @@ class JoinTest extends TableTestBase {
           unaryNode(
             "DataStreamCalc",
             streamTableNode(1),
-            term("select", "a", "b", "proctime")
+            term("select", "a", "b", "-(proctime, 3600000) AS -",
```

Comment:
Yes, you're of course right @twalthr. This is a valid rewrite and I agree with you're analysis. We have to prevent the push-down of predicates that include time indicator fields. The `DataStreamWindowJoin` needs the offsets to compute the correct window boundaries.